Rewrite botkit rocket.chat connector to TypeScript under new Botkit lib

serjk opened the pull request:
- Rewrite using new 4.10.0 and higher Botkit lib version
- Edit usage example according to the new Botkit version 
- Added new LiveChat calls 
- Made rocket.chat api utils accessing outside the connector

serjk committed:
impl livechat forward func
add utils to bot field to be able to call from outside

serjk committed:
Merge pull request #1 from serjk/feature-chat-forward

impl livechat forward func

serjk committed:
update botkit lib up to 4.10
rework adapter under the new lib
migrate adapder and utils fuctions to TypeScript
update code example

serjk committed:
Merge pull request #2 from serjk/feature-chat-forward

Adapter supports 4.10 Botkit lib version

serjk committed:
fix unit tests
remove oudated pic from readme

serjk committed:
impl LiveChat calls
extend activity from field

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/botkit-rocketchat-connector?pullRequest=8) <br/>All committers have signed the CLA.

serjk committed:
fix: lgtm alerts
add: get/update rocket.chat settings call
add: connect/diconnect calls to utils

serjk committed:
fix: remove unused import

serjk committed:
fix: code formating

serjk committed:
impl rocket chat bot worker for live chat messeges
